### ISSUE TITLE
Creating bare repository skips storage namespace validation

### ISSUE THREAD

**Issue by itaiad200** [treeverse/lakeFS#3165]
Some bad things can happen as a result, e.g.
- Creating a repo with no storage access
- Creating multiple repositories under the same namespace

**Comment by tkalir** [comment]
hi, please assign me the issue if it's available

**Comment by N-o-Z** [comment]
> hi, please assign me the issue if it's available

Done!

**Comment by tkalir** [comment]
I see that the mentioned validations (write access check and checking for existing repo at the same storage namespace) occur right now only for non-bare non-readonly repos. 
I guess that these checks should be added only for non-readonly bare repos?

**Comment by N-o-Z** [comment]
@idanovo any reason why we didn't add namespace validation on read only repos?

**Comment by idanovo** [comment]
> @idanovo any reason why we didn't add namespace validation on read only repos?

@N-o-Z 
I don't think read-only repositories are relevant to this issue; they are not the same as bare repositories. 
This issue has existed since before the announcement of the read-only repositories feature.
Am I missing something?

**Comment by idanovo** [comment]
> @idanovo any reason why we didn't add namespace validation on read-only repos?

@N-o-Z Yes- since this is a read-only repository, there is no harm in case the storage namespace we are using is already used by another repository or if we don't have write permissions for this namespace.
We can have for example one "regular" repository for this namespace and another read-only repo(s) for this namespace.
Maybe we should document this one better as it took me some time to remember we had done it.

**Comment by idanovo** [comment]
@N-o-Z https://github.com/treeverse/lakeFS/pull/8396

**Comment by arielshaqed** [comment]
I would like to understand why this is the right thing to do.  Creating a bare repo is _intentionally_ a plumbing operation, and I am afraid of limiting it, especially at this late stage.

For instance, if we make this change then restoring a repository becomes _harder_: I need to worry about whether `dummy` already exists in the repository.

This needs a few minutes of face-to-face discussion; fortunately we have some coming up this week!
